docs/rst/tutorials/cloud/wan_edge_cloud_tls/wan_edge_cloud_tls.rst

```diff
@@ -0,0 +1,390 @@
+.. _tutorials_cloud_wan_edge_cloud_tls_wan_edge_cloud_tls:
+
+
+Edge-Cloud TLS communication on WAN
+====================================
+
+.. warning::
+
+   In this example it is assumed the reader has basic knowledge of TLS concepts
+   since terms like Certificate Authority (CA), Private Key, `Rivest–Shamir–Adleman` (RSA) cryptosystem,
+   and Diffie-Hellman encryption protocol are not explained in detail.
+
+.. contents::
+    :depth: 2
+    :local:
+    :backlinks: none
+
+Background
+----------
+
+This tutorial builds on the previous one (:ref:`tutorials_cloud_wan_edge_cloud_wan_edge_cloud`),
+further showing how to secure the edge-cloud communication link with TLS protocol.
+It is recommended to follow these tutorials in order, as some concepts or installations may be already covered.
+
+.. _warning_lan_tls:
+
+.. warning::
+
+    This tutorial is intended for WAN communication.
+    However, if communication through a LAN is your only option, it is still possible to follow the tutorial by changing the ROS 2 Domain Ids so that each ROS 2 node uses a different Domain (``0`` and ``1``).
+    This way the ROS 2 nodes are logically isolated and will not discover other nodes out of their ROS 2 Domain.
+
+Following, all the elements involved in this architecture will be studied, starting with the edge robot, continuing with the controller hosted in the cloud also built as a ROS 2 node and concluding with the intermediate elements that enable communication over the Internet.
+
+The image below describes the scenario presented in this tutorial.
+
+.. figure:: /rst/figures/tutorials/cloud/edge_cloud_wan.png
```

Comment:
NIT:
change the arrow between routers or add it something to represent that it uses TCP-TLS communication (a lock or anything)
I think it is important to make clear that this secures this communication and not the rest of the network.